app/services/katello/candlepin_event_listener.rb

```diff
@@ -2,81 +2,66 @@ module Katello
   class CandlepinEventListener
     STATUS_CACHE_KEY = 'candlepin_events_status'.freeze
 
-    CandlepinEvent = Struct.new(:message_id, :subject, :content)
+    Event = Struct.new(:subject, :content)
+
+    cattr_accessor :client
 
-    @logger = ::Foreman::Logging.logger('katello/candlepin_events')
     @failed_count = 0
     @processed_count = 0
 
-    def self.run
-      initialize_listening_service
+    def self.logger
+      ::Foreman::Logging.logger('katello/candlepin_events')
+    end
+
+    def self.running?
+      @running == true && client.running?
```

Comment:
What happens if these two are in disagreement? That sounds like some kind of failure condition.

Reply:
You're exactly right. If this method returns false then the event daemon will restart the CandlepinEventListener. See Katello::EventDaemon::Monitor for how that's handled.

This would happen in a situation where the connection is somehow closed or if the listening thread in the stomp client dies, leaving an open connection but a state where messages are no longer handled